### ISSUE TITLE
Provide mapping between EFO response to terms and new OBA replacements

### ISSUE THREAD

**Issue by dosumis** [EBISPOT/efo#1436]
Generate list of all GO classes in the response to branch - this can be done easily with a query of UberGraph.

SPARQL query for xrefs to these terms => EFO terms.

Draft query here - needs some work.

TODO assign to @rays22 (needs to be added to EFO)

**Comment by rays22** [comment]
[Here is an UberGraph query](https://api.triplydb.com/s/FKY7KeMXD)
 to get the EFO terms that are subclasses of
* GO:0042221 ; # "response to chemical"
* GO:0050896 ; # "response to stimulus"
Feel free to modify it.

**Comment by dosumis** [comment]
<meta charset="utf-8"><b style="font-weight:normal;" id="docs-internal-guid-430a8bd7-7fff-367d-808e-5a7187075d74"><ol style="margin-top:0;margin-bottom:0;padding-inline-start:48px;"><li dir="ltr" style="list-style-type:decimal;font-size:11pt;font-family:Arial;color:#000000;background-color:transparent;font-weight:400;font-style:normal;font-variant:normal;text-decoration:none;vertical-align:baseline;white-space:pre;" aria-level="1"><p dir="ltr" style="line-height:1.38;margin-top:0pt;margin-bottom:0pt;" role="presentation"><span style="font-size:11pt;font-family:Arial;color:#000000;background-color:transparent;font-weight:400;font-style:normal;font-variant:normal;text-decoration:none;vertical-align:baseline;white-space:pre;white-space:pre-wrap;">We need a mapping between OBA terms and EFO/GO terms in use so that we can obsolete and replace.&nbsp; For this, we can use </span><a href="https://docs.google.com/spreadsheets/d/13qh7dLE38vMyz91oRqj6GzKjFohNazNKAJxKG6Plw1o/edit#gid=0" style="text-decoration:none;"><span style="font-size:11pt;font-family:Arial;color:#1155cc;background-color:transparent;font-weight:400;font-style:normal;font-variant:normal;text-decoration:underline;-webkit-text-decoration-skip:none;text-decoration-skip-ink:none;vertical-align:baseline;white-space:pre;white-space:pre-wrap;">Kalia&rsquo;s mapping spreadsheet</span></a><span style="font-size:11pt;font-family:Arial;color:#000000;background-color:transparent;font-weight:400;font-style:normal;font-variant:normal;text-decoration:none;vertical-align:baseline;white-space:pre;white-space:pre-wrap;"> - but note, the OBA IDs in this or NOT correct.&nbsp; Instead we can use CHEBI IDs. e.g.&nbsp;</span></p></li></ol><br /><div dir="ltr" style="margin-left:0pt;" align="left">

defined_class | defined_class_name | chemical | chemical_name |  
-- | -- | -- | -- | --
OBA:2040000 |   | CHEBI:10033 | warfarin |  

</div><br /><p dir="ltr" style="line-height:1.38;margin-top:0pt;margin-bottom:0pt;"><span style="font-size:9pt;font-family:Arial;color:#24292f;background-color:transparent;font-weight:400;font-style:normal;font-variant:normal;text-decoration:none;vertical-align:baseline;white-space:pre;white-space:pre-wrap;">As a first step, we should create a 3 column spreadsheet:&nbsp; OBA to EFO to GO mapping. </span></p></b><br class="Apple-interchange-newline">We need a mapping between OBA terms and EFO/GO terms in use so that we can obsolete and replace.  For this, we can use [Kalia’s mapping spreadsheet](https://docs.google.com/spreadsheets/d/13qh7dLE38vMyz91oRqj6GzKjFohNazNKAJxKG6Plw1o/edit#gid=0) - but note, the OBA IDs in this are NOT correct.  Instead we can use CHEBI IDs. e.g. 

defined_class
defined_class_name
chemical
chemical_name


OBA:2040000


CHEBI:10033
warfarin




Kallia’s spreadsheet has the mapping:

EFO:0009167
response to warfarin
GO:0042221
CHEBI:10033


From this we can map OBA:2040000 to 
EFO:0009167
response to warfarin
GO:0042221


As a first step, we should create a 3 column spreadsheet:  OBA to EFO to GO mapping.

**Comment by rays22** [comment]
First draft of  EFO/GO --> OBA mapping:
https://docs.google.com/spreadsheets/d/13qh7dLE38vMyz91oRqj6GzKjFohNazNKAJxKG6Plw1o/edit#gid=1861483396

**Comment by dosumis** [comment]
DEPENDENCY: OBA release needed.
- [ ] Ray to add an additional column showing obsoletion status to EFO/GO--> OBA mapping for response to chemical terms.

@zoependlington everything with SKOS:exactMatch in this sheet is good, but I think more discussion of transition is needed to ensure that labels match up and the imported terms sit in the right place in the hierarchy.

**Comment by rays22** [comment]
> * [x]  Ray to add an additional column showing obsoletion status to EFO/GO--> OBA mapping for response to chemical terms.

I have replaced the obsolete EFO classes with the replacement GO classes in the [gsheet tab.](https://docs.google.com/spreadsheets/d/13qh7dLE38vMyz91oRqj6GzKjFohNazNKAJxKG6Plw1o/edit#gid=946165155)
Now there should be no obsolete EFO/GO classes in the table.

**Comment by rays22** [comment]
@dosumis , @zoependlington ,
Is there anything else to do to for this ticket or can I close it?

**Comment by dosumis** [comment]
The aims here has to be to get OBA response_to terms seamlessly integrated into EFO.  For this I think we need a mapping sheet for role-based grouping classes as well as the ones here [gsheet tab.](https://docs.google.com/spreadsheets/d/13qh7dLE38vMyz91oRqj6GzKjFohNazNKAJxKG6Plw1o/edit#gid=946165155)

  I think what's needed it:

- import all OBA response_to terms to EFO
- Using this mapping [gsheet tab.](https://docs.google.com/spreadsheets/d/13qh7dLE38vMyz91oRqj6GzKjFohNazNKAJxKG6Plw1o/edit#gid=946165155) + one for role-based grouping classes:
   - where we have an exact match between an OBA term and an EFO term, the EFO term should be obsoleted in favour of the OBA term.  @zoependlington  - I think we've agreed this but please let me know if problematic
   - where the EFO term is more specific (broad_match in the table)? it should be retained as subclass
   - where the EFO term is broader (narrow_match in the table?) it should be retained as a superclass.
  
This would be rather painful to do by hand.  @matentzn - I'm wondering if there's a clever way to do this.  It would probably take a few attempts to the tables right and review the results.

**Comment by matentzn** [comment]
Too complicated to answer in a ticket, needs meeting

**Comment by rays22** [comment]
2022-08-08: raise this issue at the next OBA call.

**Comment by dosumis** [comment]
I think it will take a call to fix it - with some prep so that we can see plenty of examples of what.  We have the mapping table - but this is a bit confusing to look at on its own.  It would be good to have some examples where we can see the representation in  OBA and EFO side by side.  

Also before the call - @rays22 - can we get a status-update on the role-based terms? Are they now in OBA?  Is there more work to do on them?

I propose that we discuss this on the next OBA call and invite Zoe along.

**Comment by rays22** [comment]
> Also before the call - @rays22 - can we get a status-update on the role-based terms? Are they now in OBA? Is there more work to do on them?

Some role-based terms related to[response_to_chemical_with_role_stimulus](
https://github.com/obophenotype/bio-attribute-ontology/blob/master/src/patterns/data/default/response_to_chemical_with_role_stimulus.tsv) have been incorporated into OBA.

**Comment by ireneisdoomed** [comment]
We have an Open Targets issue related to this topic: [Issue #3887](https://github.com/opentargets/issues/issues/3887).

On the [response to metformin](https://platform.opentargets.org/disease/GO_1901558) disease page, we’d like to improve the user journey by linking it to the corresponding [metformin](https://platform.opentargets.org/drug/CHEMBL1431)
 drug page.

As I understand it, with the integration of OBA terms, each “response to” term can be linked to its corresponding ChEBI ID.

@zoependlington, could you clarify what steps are needed to be able to cross-reference between these terms? For this trait, we have GWAS data from the GWAS Catalog. Should we request replacing all GO terms with OBA terms?

**Comment by jamesamcl** [comment]
**Triage (2026-08-26) — status check against EFO 3.93.0**

**Summary: the mapping this ticket asks for was delivered in 2022 and partly executed. What blocks the remaining 95% is OBA coverage, not mapping effort.** Suggest re-scoping or closing as superseded by #1653 + #1543.

### What has already happened

- **87** EFO `obsolete_response to *` classes exist in `efo-edit.owl`: **73 replaced by an `OBA:2040*` term**, 14 by GO, none left without a replacement. All 73 OBA targets are imported — no dangling `IAO:0100001`.
- However, 73 of those carry the reason string *"Project to replace EFO measurement branch with enriched OBA branch"* and were obsoleted in **3.64.0 (50) and 3.74.0 (23)** — i.e. they were swept up by the **measurement** migration (#787, closed 2026-08-04) via `src/ontology/mass_obsolete_oba.py`, not by a deliberate response-to workstream.
- **94** OBA `trait in response to *` terms are imported into EFO (OLS reports ~161 upstream).

### What remains

- **130** active EFO-native `response to *` classes. Parents: `GO:0009410` ×65, `GO:0050896` ×18, `GO:0097327` ×13, `EFO:0010757` ×6, `GO:0036276` ×4, `EFO:0009170` ×4.
- I checked **all 130** against OLS for an exact OBA counterpart (`response to X` → `trait in response to X`). **Only 7 have one:**

| EFO | label | OBA | imported? |
|---|---|---|---|
| EFO:0010123 | response to antiviral drug | OBA:2040053 | yes |
| EFO:0007767 | response to calcium channel blocker | OBA:2040063 | yes |
| EFO:0920052 | response to zonisamide | OBA:2040003 | yes |
| EFO:0010969 | response to growth hormone | OBA:2040061 | **no** |
| EFO:0005417 | response to mTOR inhibitor | OBA:2040111 | **no** |
| EFO:0005657 | response to protease inhibitor | OBA:2040059 | **no** |
| EFO:0009170 | response to tyrosine kinase inhibitor | OBA:2040064 | **no** |

**123 of 130 have no OBA term to map to.** They are overwhelmingly role-based groupings ("response to beta blocker", "response to antihypertensive drug", "response to endocrine therapy") — exactly the gap #1543 was opened for. This confirms @dosumis's 2023-03-13 conclusion: no amount of mapping work clears this; it needs OBA content plus the strategy call that was never held.

### Other observations

- **No mapping file is versioned in this repo.** The mapping exists only in the Google Sheet; `mass_obsolete_oba.py` consumes an `efo_id,oba_id` CSV that was never committed. Worth landing the sheet as an SSSOM TSV so executed and pending rows are auditable.
- **15 residual `is_about` (IAO:0000136) axioms across 13 active response-to terms** remain (EFO:0005195, EFO:0005325, EFO:0005403, EFO:0010103, EFO:0011041, EFO:0020976, EFO:0600013, EFO:0600021, EFO:0600022, EFO:0600023, EFO:0600030, EFO:0600031, EFO:0600051) — see #166.
- CHEBI enrichment from #2490/#2561 reached **45/130** terms (`has_primary_input`); **80/130 still have no CHEBI link**.

### @ireneisdoomed — on the Open Targets metformin request

`OBA:2040110` *trait in response to metformin* **exists in OBA but is not imported into EFO**. EFO currently carries `GO:1901558`, whose imported stanza has only `subClassOf GO:0050896` plus a definition — no CHEBI axiom — so there is nothing in EFO today to drive a trait→drug link for that page. Two small options: import `OBA:2040110` (CHEBI-defined by construction), or extend the #2490 `has_primary_input some CHEBI` pattern to the 72 GO `response to *` classes EFO imports. The second may fit GWAS Catalog data already annotated to GO terms better.

### Suggested disposition

1. Re-title this ticket to its actual residue, or close as superseded by #1653 + #1543.
2. Commit the mapping sheet as SSSOM so it stops living off-repo.
3. Import `OBA:2040110` to unblock the Open Targets ask.
4. Import the 4 available OBA counterparts above and review those 7 EFO terms for obsoletion (labels/definitions need a human check first).
5. Clear the 15 `is_about` axioms alongside #166.

---
*This is an automated triage review by an AI agent; maintainers, please correct if anything here is wrong.*
